lib/ansible/template/__init__.py

```diff
@@ -469,56 +470,61 @@ def __getitem__(self, key):
                 if func:
                     return func
 
-                # didn't find it in the pre-built Jinja env, assume it's a former builtin and follow the normal routing path
-                leaf_key = key
-                key = 'ansible.builtin.' + key
-            else:
-                leaf_key = key.split('.')[-1]
+            key, leaf_key = get_fqcr_and_name(key)
+
+            for dummy in range(30):
```

Comment:
Curious about your thoughts on this.  I think generally I'm fine with it as is, but should we not care about depth, and instead track redirects to ensure they aren't duplicates?

I tried to look at how this is done in the plugin loader, but it's just not implemented the same here and there.

Reply:
That seems like a good idea. I'll look closer at how the plugin loader does this and tweak this.

Reply:
Updated this to loop indefinitely until the same fqcr is encountered twice. This is what the plugin loader does too - it checks if the redirect is already present in the list of redirects accumulated so far.